File: lab2/lab2.py
import math

def task1():
    n = int(input("Please enter a positive integer greater than 9: "))
    steps = 0
    sequence = [str(n)]

    while n > 9:
        sum_digits = 0
        temp = n
        while temp > 0:
            sum_digits += temp % 10
            temp //= 10

        n = sum_digits
        sequence.append(str(n))
        steps += 1

    print(" \u2192 ".join(sequence))
    print(f"Final value: {n}")
    print(f"Total steps: {steps}\n")

def task2():
    prompt = "Please enter a number between 10 and 100: "

    while True:
        n = int(input(prompt))
        if 10 <= n <= 100:
            break
        prompt = "Invalid input. Please enter a number between 10 and 100: "

    fizz = 0
    buzz = 0
    fizzbuzz = 0

    for i in range(1, n + 1):
        if i % 7 == 0:
            print(f"({i} is skipped)")
            continue
        if i % 15 == 0:
            print("FizzBuzz")
            fizzbuzz += 1
        elif i % 3 == 0:
            print("Fizz")
            fizz += 1
        elif i % 5 == 0:
            print("Buzz")
            buzz += 1
        else:
            print(i)

    print("Summary")
    print(f"Fizz count: {fizz}")
    print(f"Buzz count: {buzz}")
    print(f"FizzBuzz count: {fizzbuzz}\n")

def task3():
    n = int(input("Please enter a number between 3 and 9: "))

    for i in range(1, 2 * n):
        k = n - abs(n - i)
        for j in range(1, k + 1):
            print(j, end="")
        print()

if __name__ == "__main__":
    print("--- Task 1 ---")
    task1()
    print("--- Task 2 ---")
    task2()
    print("--- Task 3 ---")
    task3()

File: lab2/lab2.cpp
#include <iostream>
#include <string>
#include <cmath>

using namespace std;

int digitSum(int num) {
    int sum = 0;
    while (num > 0) {
        sum += num % 10;
        num /= 10;
    }
    return sum;
}

void task1() {
    int n;
    cout << "Please enter a positive integer greater than 9: ";
    while (true) {
        cin >> n;
        if (n > 9) break;
        cout << "Invalid input. Please enter a positive integer greater than 9: ";
    }

    int steps = 0;
    cout << n;

    while (n >= 10) {
        n = digitSum(n);
        steps++;
        cout << " -> " << n;
    }

    cout << endl;
    cout << "Final value: " << n << endl;
    cout << "Total steps: " << steps << endl;
}

void task2() {
    int n;
    cout << "Please enter a number between 10 and 100: ";
    while (true) {
        cin >> n;
        if (n >= 10 && n <= 100) break;
        cout << "Invalid input. Please enter a number between 10 and 100: ";
    }

    int fizzCount = 0, buzzCount = 0, fizzBuzzCount = 0;

    for (int i = 1; i <= n; ++i) {
        if (i % 7 == 0) {
            cout << "(" << i << " is skipped)" << endl;
            continue;
        }

        if (i % 3 == 0 && i % 5 == 0) {
            cout << "FizzBuzz" << endl;
            fizzBuzzCount++;
        } else if (i % 3 == 0) {
            cout << "Fizz" << endl;
            fizzCount++;
        } else if (i % 5 == 0) {
            cout << "Buzz" << endl;
            buzzCount++;
        } else {
            cout << i << endl;
        }
    }

    cout << "--- Summary ---" << endl;
    cout << "Fizz count : " << fizzCount << endl;
    cout << "Buzz count : " << buzzCount << endl;
    cout << "FizzBuzz count: " << fizzBuzzCount << endl;
}

void task3() {
    int n;
    cout << "Please enter a number between 3 and 9: ";
    while (true) {
        cin >> n;
        if (n >= 3 && n <= 9) break;
        cout << "Invalid input. Please enter a number between 3 and 9: ";
    }

    for (int i = 1; i <= 2 * n - 1; ++i) {
        int k = n - abs(n - i);
        for (int j = 1; j <= k; ++j) {
            cout << j;
        }
        cout << endl;
    }
}

int main() {
    cout << "--- Task 1 ---" << endl;
    task1();
    cout << "\n--- Task 2 ---" << endl;
    task2();
    cout << "\n--- Task 3 ---" << endl;
    task3();
    return 0;
}
